graphql/websocket.go

```diff
@@ -0,0 +1,186 @@
+package graphql
+
+import (
+	"encoding/binary"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+)
+
+const (
+	webSocketMethod         = "websocket"
+	webSocketTypeConnInit   = "connection_init"
+	webSocketTypeConnAck    = "connection_ack"
+	webSocketTypeSubscribe  = "subscribe"
+	webSocketTypeNext       = "next"
+	webSocketTypeError      = "error"
+	webSocketTypeComplete   = "complete"
+	websocketConnAckTimeOut = time.Second * 30
+)
+
+// Close codes defined in RFC 6455, section 11.7.
+const (
```

Comment:
can these, and below, be private? since hopefully the caller gets them from their client anyway.

Reply:
done